src/platform/packages/shared/kbn-esql-validation-autocomplete/src/autocomplete/commands/join/index.ts

```diff
@@ -113,12 +118,36 @@ export const suggest: CommandBaseDefinition<'join'>['suggest'] = async (
     }
 
     case 'after_on': {
-      const fields = await getColumnsByType(['any'], [], {
+      const res = await callbacks?.getColumnsFor?.({ query: 'FROM lookup_index' });
```

Comment:
Check if `res` is available as `getColumnsFor` callback is optional.

And:

```suggestion
      const res = await callbacks?.getColumnsFor?.({ query: 'FROM ' + index });
```